core/include/gnuradio-4.0/Scheduler.hpp

```diff
@@ -161,7 +163,9 @@ class SchedulerBase {
             if (const auto available = reader.available(); available > 0) {
                 const auto &input = reader.get(available);
                 outPort.streamWriter().publish([&](auto &output) { std::ranges::copy(input, output.begin()); }, available);
-                reader.consume(available);
+                if (!reader.consume(available)) {
+                    throw std::runtime_error("Failed to consume messages from message port");
```

Comment:
Since we now have the message ports in place, please avoid and replace throwing exceptions by using `emitMessage(...)` to emit error messages via the `msgOut` port.

Example code (in case of state errors) is:
```cpp
if (auto e = block.changeState(lifecycle::State::STOPPED); !e) {
    using namespace gr::message;
    this->emitMessage(this->msgOut, { { key::Kind, kind::Error }, { key::ErrorInfo, e.error().message }, { key::Location, e.error().srcLoc() } });
}
```

Reply:
Replaced the exceptions in Scheduler now with static_asserts/error messages